### PR TITLE
Ensure that the OSB GUID is never changed across API server calls

### PR DESCRIPTION
**Pull request by arschles**
Also, add some general documentation to the instance tests

Fixes https://github.com/kubernetes-incubator/service-catalog/issues/579

### CODE REVIEW

File: test/integration/clientset_test.go [hunk 1]
```diff
@@ -435,7 +435,16 @@ func TestInstanceClient(t *testing.T) {
 	}
 }
 
+func strPtr(s string) *string {
+	return &s
+}
+
 func testInstanceClient(client servicecatalogclient.Interface, name string) error {
+	const (
+		osbGUID          = "test-orig"
```
Comment (vaikas-google) [inline test/integration/clientset_test.go]: consider making this into a guid.

File: test/integration/clientset_test.go [hunk 2]
```diff
@@ -435,7 +435,16 @@ func TestInstanceClient(t *testing.T) {
...
+		osbDashboardURL  = "http://test-dashboard.com"
```
Comment (vaikas-google) [inline test/integration/clientset_test.go]: test-dashboard.example.com?

File: test/integration/clientset_test.go [hunk 3]
```diff
@@ -461,6 +474,7 @@ func testInstanceClient(client servicecatalogclient.Interface, name string) erro
 		)
 	}
 
+	// create an instance
```
Comment (vaikas-google) [inline test/integration/clientset_test.go]: I'd remove this, it's pretty clear what the next line does :)

**Comment by arschles** [comment]
thanks @vaikas-google - I've addressed your nits

**Comment by pmorie** [comment]
@arschles I think you should rebase onto #522

**Comment by arschles** [comment]
@pmorie rebased

**Comment by arschles** [comment]
CI failed on a link check. restarted

**Comment by pmorie** [comment]
@arschles legit ci failure to build integration test

**Comment by pmorie** [comment]
@arschles looks like CI failure is still legit

**Pull request merged**